### PR TITLE
fix: filter vaadin-core from versions pinning

### LINKED ISSUE #11025: [V21] Incorrect version prefix for vaadin-core after package.json update
TL;DR: the `package.json` contains version with `.` instead of `-`:

```
"@vaadin/vaadin-core": "21.0.0.alpha1"
```

### Description of the bug / feature

I’m updating docs to use `21.0.0.alpha1` and faced this problem:

```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @vaadin/vaadin-core@21.0.0.alpha1.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of 'docs'
npm ERR! notarget
```

See the diff of changes to package.json after the update:

<details>
<summary>Full diff</summary>

```diff
diff --git a/package.json b/package.json
index 97e95e2c..cb52af55 100644
--- a/package.json
+++ b/package.json
@@ -5,47 +5,47 @@
     "dependencies": {
       "@vaadin/router": "1.7.4",
       "@polymer/polymer": "3.2.0",
-      "@vaadin/vaadin-crud": "20.0.0-beta1",
-      "@vaadin/vaadin-icons": "20.0.0-beta1",
-      "@vaadin/vaadin-grid": "20.0.0-beta1",
-      "@vaadin/vaadin-split-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-combo-box": "20.0.0-beta1",
-      "@vaadin/vaadin-cookie-consent": "20.0.0-beta1",
-      "@vaadin/vaadin-upload": "20.0.0-beta1",
-      "@vaadin/vaadin-dialog": "20.0.0-beta1",
-      "@vaadin/vaadin-select": "20.0.0-beta1",
-      "@vaadin/vaadin-app-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-item": "20.0.0-beta1",
-      "@vaadin/vaadin-board": "20.0.0-beta1",
-      "@vaadin/vaadin-notification": "20.0.0-beta1",
-      "@vaadin/vaadin-charts": "20.0.0-beta1",
-      "@vaadin/vaadin-grid-pro": "20.0.0-beta1",
-      "@vaadin/vaadin-progress-bar": "20.0.0-beta1",
-      "@vaadin/vaadin-date-time-picker": "20.0.0-beta1",
-      "@vaadin/vaadin-ordered-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-login": "20.0.0-beta1",
-      "@vaadin/vaadin-button": "20.0.0-beta1",
-      "@vaadin/vaadin-date-picker": "20.0.0-beta1",
-      "@vaadin/vaadin-text-field": "20.0.0-beta1",
-      "@vaadin/vaadin-menu-bar": "20.0.0-beta1",
-      "@vaadin/vaadin-custom-field": "20.0.0-beta1",
-      "@vaadin/vaadin-form-layout": "20.0.0-beta1",
-      "@vaadin/vaadin-confirm-dialog": "20.0.0-beta1",
-      "@vaadin/vaadin-accordion": "20.0.0-beta1",
+      "@vaadin/vaadin-crud": "21.0.0-alpha2",
+      "@vaadin/vaadin-icons": "21.0.0-alpha2",
+      "@vaadin/vaadin-grid": "21.0.0-alpha2",
+      "@vaadin/vaadin-split-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-combo-box": "21.0.0-alpha2",
+      "@vaadin/vaadin-cookie-consent": "21.0.0-alpha2",
+      "@vaadin/vaadin-upload": "21.0.0-alpha2",
+      "@vaadin/vaadin-dialog": "21.0.0-alpha2",
+      "@vaadin/vaadin-select": "21.0.0-alpha2",
+      "@vaadin/vaadin-app-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-item": "21.0.0-alpha2",
+      "@vaadin/vaadin-board": "21.0.0-alpha2",
+      "@vaadin/vaadin-notification": "21.0.0-alpha2",
+      "@vaadin/vaadin-charts": "21.0.0-alpha2",
+      "@vaadin/vaadin-grid-pro": "21.0.0-alpha2",
+      "@vaadin/vaadin-progress-bar": "21.0.0-alpha2",
+      "@vaadin/vaadin-date-time-picker": "21.0.0-alpha2",
+      "@vaadin/vaadin-ordered-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-login": "21.0.0-alpha2",
+      "@vaadin/vaadin-button": "21.0.0-alpha2",
+      "@vaadin/vaadin-date-picker": "21.0.0-alpha2",
+      "@vaadin/vaadin-text-field": "21.0.0-alpha2",
+      "@vaadin/vaadin-menu-bar": "21.0.0-alpha2",
+      "@vaadin/vaadin-custom-field": "21.0.0-alpha2",
+      "@vaadin/vaadin-form-layout": "21.0.0-alpha2",
+      "@vaadin/vaadin-confirm-dialog": "21.0.0-alpha2",
+      "@vaadin/vaadin-accordion": "21.0.0-alpha2",
       "@polymer/iron-list": "3.1.0",
-      "@vaadin/vaadin-list-box": "20.0.0-beta1",
-      "@vaadin/vaadin-checkbox": "20.0.0-beta1",
-      "@vaadin/vaadin-details": "20.0.0-beta1",
+      "@vaadin/vaadin-list-box": "21.0.0-alpha2",
+      "@vaadin/vaadin-checkbox": "21.0.0-alpha2",
+      "@vaadin/vaadin-details": "21.0.0-alpha2",
       "@polymer/iron-icon": "3.0.1",
-      "@vaadin/vaadin-time-picker": "20.0.0-beta1",
-      "@vaadin/vaadin-context-menu": "20.0.0-beta1",
-      "@vaadin/vaadin-radio-button": "20.0.0-beta1",
-      "@vaadin/vaadin-tabs": "20.0.0-beta1",
-      "@vaadin/vaadin-lumo-styles": "20.0.0-beta1",
-      "@vaadin/vaadin-material-styles": "20.0.0-beta1",
-      "@vaadin/vaadin-rich-text-editor": "20.0.0-beta1",
-      "@vaadin/vaadin-avatar": "20.0.0-beta1",
-      "@vaadin/vaadin-messages": "20.0.0-beta1",
+      "@vaadin/vaadin-time-picker": "21.0.0-alpha2",
+      "@vaadin/vaadin-context-menu": "21.0.0-alpha2",
+      "@vaadin/vaadin-radio-button": "21.0.0-alpha2",
+      "@vaadin/vaadin-tabs": "21.0.0-alpha2",
+      "@vaadin/vaadin-lumo-styles": "21.0.0-alpha2",
+      "@vaadin/vaadin-material-styles": "21.0.0-alpha2",
+      "@vaadin/vaadin-rich-text-editor": "21.0.0-alpha2",
+      "@vaadin/vaadin-avatar": "21.0.0-alpha2",
+      "@vaadin/vaadin-messages": "21.0.0-alpha2",
       "lit": "2.0.0-rc.1",
       "@polymer/iron-a11y-announcer": "3.0.2",
       "@polymer/iron-a11y-keys-behavior": "3.0.1",
@@ -58,13 +58,18 @@
       "@polymer/iron-resizable-behavior": "3.0.1",
       "@polymer/iron-scroll-target-behavior": "3.0.1",
       "@webcomponents/shadycss": "1.9.6",
-      "@vaadin/vaadin-control-state-mixin": "20.0.0-beta1",
+      "@vaadin/vaadin-control-state-mixin": "21.0.0-alpha2",
       "@vaadin/vaadin-development-mode-detector": "2.0.4",
-      "@vaadin/vaadin-element-mixin": "20.0.0-beta1",
-      "@vaadin/vaadin-list-mixin": "20.0.0-beta1",
-      "@vaadin/vaadin-overlay": "20.0.0-beta1",
-      "@vaadin/vaadin-themable-mixin": "20.0.0-beta1",
-      "@vaadin/vaadin-usage-statistics": "2.1.0"
+      "@vaadin/vaadin-element-mixin": "21.0.0-alpha2",
+      "@vaadin/vaadin-list-mixin": "21.0.0-alpha2",
+      "@vaadin/vaadin-overlay": "21.0.0-alpha2",
+      "@vaadin/vaadin-themable-mixin": "21.0.0-alpha2",
+      "@vaadin/vaadin-usage-statistics": "2.1.0",
+      "@vaadin/vaadin-core-shrinkwrap": "21.0.0-alpha1",
+      "@vaadin/vaadin-shrinkwrap": "21.0.0-alpha1",
+      "@vaadin/vaadin-template-renderer": "21.0.0-alpha2",
+      "@vaadin/vaadin-virtual-list": "21.0.0-alpha2",
+      "@vaadin/vaadin-core": "21.0.0.alpha1"
     },
     "devDependencies": {
       "compression-webpack-plugin": "4.0.1",
@@ -92,7 +97,7 @@
       "ts-loader": "8.0.12",
       "fork-ts-checker-webpack-plugin": "6.2.1"
     },
-    "hash": "0c81b06dbbedba1f96e99c389b5e7a37cecd8eaff1b64e7b8f9c7aeb071c32ee"
+    "hash": "ce15191bddb8096d9b7f0b0955d12ed12b8e6415ad58a6171e721a02ddae61d6"
   },
   "dependencies": {
     "@polymer/iron-a11y-announcer": "3.0.2",
@@ -111,57 +116,62 @@
     "@vaadin/flow-frontend": "./target/flow-frontend",
     "@vaadin/form": "./target/flow-frontend/form",
     "@vaadin/router": "1.7.4",
-    "@vaadin/vaadin-accordion": "20.0.0-beta1",
-    "@vaadin/vaadin-app-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-avatar": "20.0.0-beta1",
-    "@vaadin/vaadin-board": "20.0.0-beta1",
-    "@vaadin/vaadin-button": "20.0.0-beta1",
-    "@vaadin/vaadin-charts": "20.0.0-beta1",
-    "@vaadin/vaadin-checkbox": "20.0.0-beta1",
-    "@vaadin/vaadin-combo-box": "20.0.0-beta1",
-    "@vaadin/vaadin-confirm-dialog": "20.0.0-beta1",
-    "@vaadin/vaadin-context-menu": "20.0.0-beta1",
-    "@vaadin/vaadin-control-state-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-cookie-consent": "20.0.0-beta1",
-    "@vaadin/vaadin-crud": "20.0.0-beta1",
-    "@vaadin/vaadin-custom-field": "20.0.0-beta1",
-    "@vaadin/vaadin-date-picker": "20.0.0-beta1",
-    "@vaadin/vaadin-date-time-picker": "20.0.0-beta1",
-    "@vaadin/vaadin-details": "20.0.0-beta1",
+    "@vaadin/vaadin-accordion": "21.0.0-alpha2",
+    "@vaadin/vaadin-app-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-avatar": "21.0.0-alpha2",
+    "@vaadin/vaadin-board": "21.0.0-alpha2",
+    "@vaadin/vaadin-button": "21.0.0-alpha2",
+    "@vaadin/vaadin-charts": "21.0.0-alpha2",
+    "@vaadin/vaadin-checkbox": "21.0.0-alpha2",
+    "@vaadin/vaadin-combo-box": "21.0.0-alpha2",
+    "@vaadin/vaadin-confirm-dialog": "21.0.0-alpha2",
+    "@vaadin/vaadin-context-menu": "21.0.0-alpha2",
+    "@vaadin/vaadin-control-state-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-cookie-consent": "21.0.0-alpha2",
+    "@vaadin/vaadin-crud": "21.0.0-alpha2",
+    "@vaadin/vaadin-custom-field": "21.0.0-alpha2",
+    "@vaadin/vaadin-date-picker": "21.0.0-alpha2",
+    "@vaadin/vaadin-date-time-picker": "21.0.0-alpha2",
+    "@vaadin/vaadin-details": "21.0.0-alpha2",
     "@vaadin/vaadin-development-mode-detector": "2.0.4",
-    "@vaadin/vaadin-dialog": "20.0.0-beta1",
-    "@vaadin/vaadin-element-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-form-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-grid": "20.0.0-beta1",
-    "@vaadin/vaadin-grid-pro": "20.0.0-beta1",
-    "@vaadin/vaadin-icons": "20.0.0-beta1",
-    "@vaadin/vaadin-item": "20.0.0-beta1",
-    "@vaadin/vaadin-list-box": "20.0.0-beta1",
-    "@vaadin/vaadin-list-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-login": "20.0.0-beta1",
-    "@vaadin/vaadin-lumo-styles": "20.0.0-beta1",
-    "@vaadin/vaadin-material-styles": "20.0.0-beta1",
-    "@vaadin/vaadin-menu-bar": "20.0.0-beta1",
-    "@vaadin/vaadin-messages": "20.0.0-beta1",
-    "@vaadin/vaadin-notification": "20.0.0-beta1",
-    "@vaadin/vaadin-ordered-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-overlay": "20.0.0-beta1",
-    "@vaadin/vaadin-progress-bar": "20.0.0-beta1",
-    "@vaadin/vaadin-radio-button": "20.0.0-beta1",
-    "@vaadin/vaadin-rich-text-editor": "20.0.0-beta1",
-    "@vaadin/vaadin-select": "20.0.0-beta1",
-    "@vaadin/vaadin-split-layout": "20.0.0-beta1",
-    "@vaadin/vaadin-tabs": "20.0.0-beta1",
-    "@vaadin/vaadin-text-field": "20.0.0-beta1",
-    "@vaadin/vaadin-themable-mixin": "20.0.0-beta1",
-    "@vaadin/vaadin-time-picker": "20.0.0-beta1",
-    "@vaadin/vaadin-upload": "20.0.0-beta1",
+    "@vaadin/vaadin-dialog": "21.0.0-alpha2",
+    "@vaadin/vaadin-element-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-form-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-grid": "21.0.0-alpha2",
+    "@vaadin/vaadin-grid-pro": "21.0.0-alpha2",
+    "@vaadin/vaadin-icons": "21.0.0-alpha2",
+    "@vaadin/vaadin-item": "21.0.0-alpha2",
+    "@vaadin/vaadin-list-box": "21.0.0-alpha2",
+    "@vaadin/vaadin-list-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-login": "21.0.0-alpha2",
+    "@vaadin/vaadin-lumo-styles": "21.0.0-alpha2",
+    "@vaadin/vaadin-material-styles": "21.0.0-alpha2",
+    "@vaadin/vaadin-menu-bar": "21.0.0-alpha2",
+    "@vaadin/vaadin-messages": "21.0.0-alpha2",
+    "@vaadin/vaadin-notification": "21.0.0-alpha2",
+    "@vaadin/vaadin-ordered-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-overlay": "21.0.0-alpha2",
+    "@vaadin/vaadin-progress-bar": "21.0.0-alpha2",
+    "@vaadin/vaadin-radio-button": "21.0.0-alpha2",
+    "@vaadin/vaadin-rich-text-editor": "21.0.0-alpha2",
+    "@vaadin/vaadin-select": "21.0.0-alpha2",
+    "@vaadin/vaadin-split-layout": "21.0.0-alpha2",
+    "@vaadin/vaadin-tabs": "21.0.0-alpha2",
+    "@vaadin/vaadin-text-field": "21.0.0-alpha2",
+    "@vaadin/vaadin-themable-mixin": "21.0.0-alpha2",
+    "@vaadin/vaadin-time-picker": "21.0.0-alpha2",
+    "@vaadin/vaadin-upload": "21.0.0-alpha2",
     "@vaadin/vaadin-usage-statistics": "2.1.0",
     "@webcomponents/shadycss": "1.9.6",
     "date-fns": "^2.16.1",
     "lit": "2.0.0-rc.1",
     "promise-file-reader": "^1.0.3",
-    "vanilla-colorful": "^0.5.3"
+    "vanilla-colorful": "^0.5.3",
+    "@vaadin/vaadin-core-shrinkwrap": "21.0.0-alpha1",
+    "@vaadin/vaadin-shrinkwrap": "21.0.0-alpha1",
+    "@vaadin/vaadin-template-renderer": "21.0.0-alpha2",
+    "@vaadin/vaadin-virtual-list": "21.0.0-alpha2",
+    "@vaadin/vaadin-core": "21.0.0.alpha1"
   },
   "devDependencies": {
     "@types/validator": "13.1.0",
```

</details>

### Minimal reproducible example

1. `git clone https://github.com/vaadin/docs` 
2. `git checkout lit2-update`
3. Update version in pom.xml to be `21.0.0.alpha1`
4. Run `mvn`

### Expected behavior

The version should look like this (using `-`):

```
"@vaadin/vaadin-core": "21.0.0-alpha1"
```

### Actual behavior

The version looks like this (using `.`):

```
"@vaadin/vaadin-core": "21.0.0.alpha1"
```

### Versions:
    - Vaadin / Flow version: 21.0.0.alpha1

### PR DESCRIPTION
**Pull request by pleku**
Fixes #11025

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java [hunk 1]
```diff
@@ -75,6 +75,7 @@
     static final String VAADIN_DEP_KEY = "vaadin";
     static final String HASH_KEY = "hash";
     static final String DEV_DEPENDENCIES = "devDependencies";
+    static final String VAADIN_CORE_NPM_PACKAGE = "@vaadin/vaadin-core";
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java:78]: Wouln't this be more a part of VersionsJsonConverter as no NodeUpdater uses it?

Comment (pleku) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java:78]: Yes but I needed it somewhere to share it with the tests - does it really matter ?

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java:78]: Having it in VersionsJsonConverter would have been just as usable in tests and needing an import in the only place using it (apart from tests) seems like extra run around when having to read the code some time later.

Comment (pleku) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java]: Done. When I look at the classses, I feel that the best solution would be one location that has all the related constants as now some of them are here and some there. But that refactoring can be done later

**Review by caalador — APPROVED** [review]

**Pull request merged**